includes/Feed/FeedConfigurationDetection.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace SkyVerge\WooCommerce\Facebook\Feed;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use Error;
+use SkyVerge\WooCommerce\Facebook\Utilities\Heartbeat;
+use SkyVerge\WooCommerce\Facebook\Feed\FeedFileHandler;
+
+/**
+ * A class responsible detecting feed configuration.
+ */
+class FeedConfigurationDetection {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_action( Heartbeat::DAILY, array( $this, 'get_data_source_feed_tracker_info' ) );
+	}
+
+	/**
+	 * Get config settings for feed-based sync for WooCommerce Tracker.
+	 *
+	 * Gets various settings related to the feed, and data about recent uploads.
+	 * This is formatted into an array of keys/values, and saved to a transient for inclusion in tracker snapshot.
+	 *
+	 * @throws Error
+	 * @since x.x.x
+	 * @return Array Key-value array of various configuration settings.
+	 */
+	public function get_data_source_feed_tracker_info() {
+		$integration         = facebook_for_woocommerce()->get_integration();
+		$graph_api           = $integration->get_graph_api();
+		$integration_feed_id = $integration->get_feed_id();
+		$catalog_id          = $integration->get_product_catalog_id();
+
+		$info = array();
+
+		// No catalog id. Most probably means that we don't have a valid connection.
+		if ( '' === $catalog_id ) {
+			facebook_for_woocommerce()->log( 'No catalog ID!' );
+			return false;
+		}
+
+		// Get all feeds configured for the catalog.
+		try {
+			$feed_nodes = $this->get_feed_nodes_for_catalog( $catalog_id, $graph_api );
+		} catch ( \Error $th ) {
+			throw $th;
+		}
+
+		$info['feed_count'] = count( $feed_nodes );
+
+		// Check if the catalog has any feed configured.
+		if ( empty( $feed_nodes ) ) {
+			facebook_for_woocommerce()->log( 'No feed nodes!' );
+			return false;
+		}
+
+		// Determine which is the most active feed (recently updated).
```

Comment:
`most actvive` !== `recently updated`

If you have two feeds: one that was updated 1h ago and another one that was updated 23h ago it may also mean that the 23h ago will be updated +1h ( if this is a daily schedule ). Out of those two, you don't know which one would be the correct one.

The first thing that would need to happen is that you need to evaluate the feed indicated by the `$integration_feed_id` which now is not used. If that one is not set or has no corresponding config then we should proceed to evaluate the remaining items.

The items should only be evaluated if the 'url' in the `schedule` section matches the one used by our site. If the `url` does not match then this means that this feed config is not related to the site where the plugin is installed.

Reply:
Thanks for the tips! I've tweaked the selection of the "active" feed:

- If a feed config ID matches `$integration_feed_id`, it's selected as the "active" feed.

Note: the "active" feed in this case just means the one that we will use for tracker props. If the majority of sites have multiple feeds set up, we will not be able to draw many conclusions. This code assumes the normal case is a single feed config. Does that make sense? I just want to capture useful info to help us understand if the feed mechanism is being used on sites. In the case of multiple feeds configured, the most recent upload seems a good heuristic for selecting which config to track.

Also, for the most recent upload of the "active" feed, we now track `url-matches-site-endpoint`. If this is false, then something else is going on (e.g. site hostname has changed, secret has been regenerated, or the merchant has some complex config where their feed file is hosted elsewhere like a CDN or AWS bucket).